ray-operator/controllers/ray/common/pod.go

```diff
@@ -139,6 +143,44 @@ func DefaultHeadPodTemplate(instance rayiov1alpha1.RayCluster, headSpec rayiov1a
 	return podTemplate
 }
 
+// getAutoscalerImage determines the default autoscaler image
+func getAutoscalerImage(rayHeadImage string, rayVersion string) string {
+	if autoscalerSupportIsStable(rayVersion) {
+		// For Ray versions >= 2.0.0, use the Ray head's image to run the autoscaler.
+		return rayHeadImage
+	} else {
+		// For older Ray versions, use the Ray 2.0.0 image to run the autoscaler.
+		return FallbackDefaultAutoscalerImage
+	}
+}
+
+// Determine if autoscaler support is stable in the given rayVersion.
+// Return false exactly when the major version is successfully parsed and less than 2.
+// Example rayVersion inputs that return true: "2.0.0", "2.0", "2", "2.0.0rc1", "nightly", "latest", "unknown".
+// Example inputs that return false: "1.13.0", "1.12", "1".
+func autoscalerSupportIsStable(rayVersion string) bool {
+	// Try to determine major version by extracting everything that comes before the first "."
+	firstDotIndex := strings.Index(rayVersion, ".")
```

Comment:
minor: In our downstream code, we have similar version specific configuration. for those cases, we use `hashicorp/go-version` to directly compare versions if the version follow semantic versioning. 

```go
import "github.com/hashicorp/go-version"

// LessThanVersion tests if version1 is less than version2.
func LessThanVersion(version1 string, version2 string) (bool, error) {
	v1, err := version.NewVersion(version1)
	if err != nil {
		return false, err
	}
	v2, err := version.NewVersion(version2)
	if err != nil {
		return false, err
	}

	return v1.LessThan(v2), nil
}
```

Reply:
Thanks for the tip! I'll keep it in mind to use that library for version comparisons in the future.

For now, I prefer to implement the logic directly, as I'd like to be super explicit with what we're doing with ill-formatted version strings.